content/api/commands/session.md

```diff
@@ -0,0 +1,273 @@
+
+- information about the currently authenticated user
+  - permissions
+  - feature toggles
+  - location
+  - user behavior
+- settings that dictate how your application should appear or behave
+  - language
+  - currency
+  - color scheme
+  - accessibility features
+  - other UI preferences
+- form data
+
+## Usage & Syntax
+
+A session must be first be defined:
+
+##### **Define the Session**
+
+```js
+cy.session(name, setupFn, sessionOptions)
+```
+
+Once defined, it can be used later:
+
+##### **Use the (Previously Defined) Session**
+
+```js
+cy.session(name)
+```
+
+### Correct Usage
+
+```js
+cy.session(
+  'Peter Pan',
+  () => {
+    // code to generate a session goes here
+  },
+  {
+    validate: () => {
+      // code that returns `true` if the session is still valid
+    },
+    exclude: {
+      cookies: [
+        // cookies to exclude
+      ],
+      localStorage: [
+        // local storage items to exclude
+      ],
+    },
+  }
+)
+```
+
+### Arguments
+
+#### <Icon name="angle-right"></Icon> name (`string`)
+
+`name` is the name you would like to refer to this session as. This could be the name of the user or type of user you're defining or using - like "Adam" or "Administrator".
+
+#### <Icon name="angle-right"></Icon> setupFn (`Function`)
+
+`setupFn` is the function that, when executed, updates the browser's storage, cookies, or both, with data that Cypress can preserve so you can use the session again later without having to execute again - unless it's no longer valid (see [Validate]()).
+
+#### <Icon name="angle-right"></Icon> sessionOptions (`SessionOptions`)
+
+`sessionOptions` contains two properties:
+
+- `validate` (`Function`)
+  - returns `true`: The session is valid and will be used.
+  - returns `false`: Cypress will execute the `setupFn` to generate a new session to use.
+- `exclude` (`object`) - contains properties `cookies` and `localStorage` for specifying items that you don't want Cypress to preserve.
+
+See [SessionOptions]()
+
+### Yields
+
+`session` yields session data (`SessionData`), which is an `object` with properties `cookies` and `localStorage`. See [SessionData](#sessiondata).
+
+## Examples
+
+### Define a Session
+
+#### Name & Setup Function
+
+The `name` should represent the kind of session you're defining. For example, if you're setting up a session for a user with read-only permissions, then you might choose the name "read-only".
+The contents of `setupFn` should be the steps to generate the session you want to preserve for later use.
+
+```js
+cy.session('read-only', () => { signIn('Adam', 'Pa$$w0Rd') })
+
+function signIn(username, password) {
+  cy.intercept('POST', '/auth').as('signIn')
+  cy.visit('/sign-in')
+  cy.get('.username').type(username)
+  cy.get('.password').type(password)
+  cy.contains('.button', 'Sign In').click()
+  cy.wait('@signIn')
+})
+```
+
+#### SessionOptions
+
+The third and last argument you can optionally pass to `session` is `SessionOptions`.
```

Comment:
Just probably wouldn't hardcode the arg position in the docs, in case we change the API, add another argument so it's not the last, we may not catch this to update it later.

```suggestion
You can optionally pass `SessionOptions` to `session`.
```

Reply:
Good point. Thanks!